azure/Guidelines.md

```diff
@@ -88,80 +86,53 @@ All Azure APIs **MUST** use explicit versioning. The Microsoft REST API guidelin
-
-At a high level, changes to the contract of an API constitute a breaking change. Changes that impact backwards compatibility of an API is also considered a breaking change. Anything that would violate the Principle of Least Astonishment is considered a breaking change in Azure. Below are some concrete examples of what constitutes a breaking change. In the below breaking change scenarios, the API version must be changed.
-
-#### Existing property is removed
-
-If a property called `foo` that was present in v1 of the API needs to be removed, it must be done in a newer API version.
-
-#### New property added to response
-
-If a new property/field is added to the response of an API, the GET-PUT pipeline will be broken. Consider the case where a customer updates the value of a new property "A" from the Azure portal. Another customer does a GET of this resource using the SDK. The SDK will ignore the property since it does not understand it. From the SDK, the customer does a PUT using the model that was returned from the GET. This will overwrite the change made by the first customer from the portal.
-
-#### New required property added to request
+### API Changes that require a version change
 
-If a new property is made required in the request body, clients will have no way to set this and the request will fail.
+Azure services **MUST** update the version number of their API whenever there is a change to the API, no matter how small.  Customers will "lock the API version" so that their code does not fail when the service introduces new features.  They rely on the fact that an API version is a contract with the services that will never change.
 
-#### Property name has changed
+A breaking change is any change in the API that may cause client or service code making the API call to fail. Obvious examples of such a change are the removal of an endpoint, adding or removing a required field or changing the format of the body (from XML to JSON for example). Even though we recommend clients ignore new fields, there are many libraries and clients that fail when new fields are introduced.
 
-Note that this is implied by the requirement that adding/and removing properties are breaking changes, but in some ways worse, since it leads to the possibility of reusing a property name. Even with an API version change, this change is discouraged because it creates documentation and cognitive challenges.
+There are three reasons why a service may issue a breaking change:
 
-#### Property type has changed
+1. To remove a security vulnerability.
+2. To comply with regulatory requirements.
+3. To deprecate a feature of the service.
```

Comment:
I'm still a little unclear on this guidance, and I'm wondering if restructuring it a bit would make it easier to learn from this document.  

The big thing I'd like to clarify is how we use the term "breaking change" or whether we use it at all -- I know some are in favor of calling these "service evolutions" instead, because they won't technically break customers if you correctly follow the deprecation policy.  We seem to have the following multiple meanings of breaking:
- all changes are "breaking" (and therefore require an updated version),
- changes that impact compatibility with prior APIs are "breaking", but some additions don't impact compatibility so are "non-breaking"
- the changes that impact backward compatibility can be done as long as they are not actually "breaking" (i.e. are service evolutions that follow the deprecation policy).
- changes that will actually "break" customers (i.e. affect backward compatibility with prior API versions) can be done only for security and regulatory reasons, subject to REST API Board approval.  (Question: these can affect existing API versions, so would be done without a version bump?)

If this is the case, perhaps restructuring the guidance something like the following would make it a little more clear?

1. All changes to the API require a version change (because all changes are breaking, if this is how we choose to define breaking)
2. Some changes are allowed between versions, namely, those that don't impact backward compatibility
    a. Large additions to the API require major version updates
    b. Small additions to the APIs require minor version updates
3. Changes that would impact backward compatibility with any prior versions are allowed only under the three conditions, and subject to API Board approval
    a. for conditions 1 & 2 (security & regulatory requirements), services may actually break compatibility within an existing API version.
    b. for condition 3 (deprecation), you must follow the API deprecation policy below so that no customer is broken.